7-extract_plier_pathways.R

```diff
@@ -147,8 +200,51 @@ for(seed_index in 1:length(norm.train.files)) {
     names(plier_results_list[[i]]) <- norm_methods
   }
   
-  # TODO NOW DO COMPARISON METRIC  
-
+  # Set  array and seq silver standards for Jaccard comparison metric
+  # TODO what are best settings for silver standard?
+  array_silver <- plier_results_list[["0"]][["log"]][["summary"]] %>%
+    filter(FDR < 0.05) %>%
+    pull(pathway) %>%
+    unique()
+  seq_silver <- plier_results_list[["100"]][["log"]][["summary"]] %>%
+    filter(FDR < 0.05) %>%
+    pull(pathway) %>%
+    unique()
+  
+  # Check that silver standard pathways have non-zero length
+  if (length(array_silver) > 0 & length(seq_silver) > 0) {
+  
+    # Return pathway comparison for appropriate level of PLIER results list
+    jaccard_list[[seed_index]] <- purrr::modify_depth(plier_results_list, 2,
+                                                      function(x) return_plier_jaccard(x, array_silver, seq_silver))
+    
+  } else {
+    
+    message(str_c("PLIER: Silver standard array or seq significant pathways has zero length",
+                  seed_index, percent_seq, normalization_method,
+                  length(array_silver), length(seq_silver),
+                  sep = " "))
+    
+  }
 }
 
-# TODO PLOT THAT
+if (length(jaccard_list) > 0) {
+  
+  # melt jaccard list elements into one data frame
+  jaccard_df <- reshape2::melt(data = jaccard_list,
+                               id.vars = c("silver", "n_silver", "n_test",
+                                           "n_intersect", "n_union",
+                                           "n_common_genes", "k"),
+                               value.name = "jaccard") %>%
+    rename("nmeth" = "L3", # normalization method
+           "pseq" = "L2", # percentage RNA-seq
+           "seed_index" = "L1")
+  
+  readr::write_tsv(x = jaccard_df,
+                   path = file.path(res.dir,
+                                    str_c(file_identifier, "_PLIER_jaccard.tsv")))
+  
+  # TODO PLOT THAT
+}
+  
+  
```

Comment:
The fact that there's a no new line character here seems odd!

Reply:
Funky! I updated with styler and tried to resolve any weirdness